rosette/src/Import.cc

```diff
@@ -24,150 +24,359 @@ extern pOb NILmeta;
+    case ObjectCodePB::OT_REQUEST_EXPR: {
+        ObjectCodePB::RequestExpr * pbob = ob->mutable_request_expr();
+        retval = RequestExpr::create(
+                    SYMBOL(pbob->target().symbol().name().c_str()),
+                    (TupleExpr *)createRosetteObject( pbob->mutable_msg() ));
         break;
-    case ObjectCodePB::OT_SEND_EXPR:
-        retval = SendExpr::create( SYMBOL(ob.send_expr().target().symbol().name().c_str()), NULL);
+    }
+
+    case ObjectCodePB::OT_SEND_EXPR: {
+        ObjectCodePB::SendExpr * pbob = ob->mutable_send_expr();
+        retval = SendExpr::create(
+                    SYMBOL(pbob->target().symbol().name().c_str()),
+                    (TupleExpr *)createRosetteObject( pbob->mutable_msg() ));
         break;
-    case ObjectCodePB::OT_SEQ_EXPR:
-        retval = SeqExpr::create(NULL);
+    }
+
+    case ObjectCodePB::OT_SEQ_EXPR: {
+        ObjectCodePB::SeqExpr * pbob = ob->mutable_seq_expr();
+        retval = SeqExpr::create( (Tuple *)createRosetteObject(pbob->mutable_subexprs()) );
         break;
-    case ObjectCodePB::OT_SET_EXPR:
-        retval = SetExpr::create( SYMBOL(ob.set_expr().trgt().symbol().name().c_str()), NULL);
+    }
+
+    case ObjectCodePB::OT_SET_EXPR: {
+        ObjectCodePB::SetExpr * pbob = ob->mutable_set_expr();
+        retval = SetExpr::create( createRosetteObject(pbob->mutable_trgt()),
+                                    createRosetteObject(pbob->mutable_val()) );
         break;
-    case ObjectCodePB::OT_STD_METHOD:
-        retval = StdMthd::create(NULL, NULL, NULL);
+    }
+
+    case ObjectCodePB::OT_STD_METHOD: {
+        ObjectCodePB::StdMthd * pbob = ob->mutable_std_mthd();
+        retval = StdMthd::create( (Code *)createRosetteObject(pbob->mutable_code()),
+                                    createRosetteObject(pbob->mutable_id()),
+                                    createRosetteObject(pbob->mutable_source()));
         break;
+    }
+
     case ObjectCodePB::OT_SYMBOL:
-        retval = SYMBOL(ob.symbol().name().c_str());
+        retval = SYMBOL(ob->symbol().name().c_str());
         break;
-    case ObjectCodePB::OT_TBL_OBJECT:
+
+    case ObjectCodePB::OT_TBL_OBJECT: {
+        ObjectCodePB::TblObject * pbob = ob->mutable_tbl_object();
         retval = TblObject::create();
         break;
-    case ObjectCodePB::OT_TEMPLATE:
-        retval = Template::create(NIL, NILmeta, NILpattern);
-        break;
-    case ObjectCodePB::OT_TUPLE:
-        retval = Tuple::create();
-        break;
-    case ObjectCodePB::OT_TUPLE_EXPR:
-        retval = TupleExpr::create();
+    }
+
+    case ObjectCodePB::OT_TEMPLATE: {
+        ObjectCodePB::Template * pbob = ob->mutable_template_();
+        retval = Template::create( (Tuple *)createRosetteObject(pbob->mutable_keytuple()),
+                                    createRosetteObject(pbob->mutable_pat()),
+                                    (CompoundPattern *)createRosetteObject(pbob->mutable_keymeta()) );
```

Comment:
Why do you cast `keymeta` to `CompoundPattern*`?

Reply:
Because I inadvertently had the keymeta and pat parameters swapped in the call to Template::create(). I've fixed it.